sunpy/net/dataretriever/sources/tests/test_goes_suvi.py

```diff
@@ -19,208 +15,77 @@ def suvi_client():
-    This also checks if the mock is working well.
-    """
-    start = '2019/05/25 00:50'
-    end = '2019/05/25 00:52'
-    wave = 94 * u.Angstrom
+@pytest.mark.parametrize(('start', 'end', 'wave', 'level', 'expected_num_files'), [
+    ('2019/05/25 00:50', '2019/05/25 00:54', 94, '1b', 6),
+    ('2019/05/25 00:50', '2019/05/25 00:54', 304, '2', 1),
+])
+def test_combined_search(suvi_client, start, end, wave, level, expected_num_files):
     goes_sat = a.goes.SatelliteNumber.sixteen
-    tr = a.Time(start, end)
-    qr1 = suvi_client.search(tr, a.Instrument.suvi, a.Wavelength(wave), goes_sat, a.Level(2))
-
-    # Mock QueryResponse object
-    mock_qr = mock_query_object(suvi_client)
-
-    # Compare if two objects have the same attribute
+    qresponse = suvi_client.search(a.Time(start, end), a.Wavelength(wave * u.Angstrom), goes_sat, a.Level(level))
+    print("Query Response Length:", len(qresponse))
+    # print("Query Response Table:")
+    # print(qresponse.table)
+    print(f"Start: {start}, End: {end}, Wavelength: {wave}, Level: {level}")
+    assert len(qresponse) == expected_num_files
 
-    mock_qr = mock_qr[0]
-    qr = qr1[0]
-
-    assert mock_qr['Source'] == qr['Source']
-    assert mock_qr['Provider'] == qr['Provider']
-    assert mock_qr['Physobs'] == qr['Physobs']
-    assert mock_qr['Instrument'] == qr['Instrument']
-    assert mock_qr['url'] == qr['url']
-
-    assert qr1['Start Time'] == Time("2019-05-25T00:52:00.000")
-    assert qr1['End Time'] == Time("2019-05-25T00:56:00.000")
-
-    with tempfile.TemporaryDirectory() as tmpdirname:
-        download_list = suvi_client.fetch(qr1, path=tmpdirname)
-    assert len(download_list) == len(qr1)
 
 
 @pytest.mark.remote_data
-@pytest.mark.parametrize(('start', 'end', 'wave', 'expected_num_files'),
-                         [('2019/05/25 00:50', '2019/05/25 00:52', 94, 1),
-                          ('2019/05/25 00:50', '2019/05/25 00:52', 131, 1),
-                          ('2019/05/25 00:50', '2019/05/25 00:52', 171, 1),
-                          ('2019/05/25 00:50', '2019/05/25 00:52', 195, 1),
-                          ('2019/05/25 00:50', '2019/05/25 00:52', 284, 1),
-                          ('2019/05/25 00:50', '2019/05/25 00:52', 304, 1)]
-                         )
-def test_get_url_for_time_range_level2(suvi_client, start, end, wave, expected_num_files):
-    goes_sat = a.goes.SatelliteNumber.sixteen
-    qresponse = suvi_client.search(a.Time(start, end), a.Wavelength(wave * u.Angstrom), goes_sat, a.Level(2))
-    urls = [i['url'] for i in qresponse]
-    assert isinstance(urls, list)
-    assert len(urls) == expected_num_files
+def test_get_all_wavelengths_level2(suvi_client):
+    """Check retrieval for all wavelengths without specifying one."""
+    qresponse = suvi_client.search(a.Time('2019/05/25 00:50', '2019/05/25 00:52'),
+                                   a.goes.SatelliteNumber.sixteen, a.Level(2))
+    assert len(qresponse) == 6
```

Comment:
We should check the wavelengths here.

Reply:
------------------------------------------------------------------------------ Captured stdout call -------------------------------------------------------------------------------
       Start Time               End Time        Instrument Physobs Source Provider SatelliteNumber Level Wavelength
                                                                                                          Angstrom
----------------------- ----------------------- ---------- ------- ------ -------- --------------- ----- ----------
2019-05-25 00:52:00.000 2019-05-25 00:56:00.000       SUVI    flux   GOES     NOAA              16     2       94.0
2019-05-25 00:52:00.000 2019-05-25 00:56:00.000       SUVI    flux   GOES     NOAA              16     2      131.0
2019-05-25 00:52:00.000 2019-05-25 00:56:00.000       SUVI    flux   GOES     NOAA              16     2      171.0
2019-05-25 00:52:00.000 2019-05-25 00:56:00.000       SUVI    flux   GOES     NOAA              16     2      195.0
2019-05-25 00:52:00.000 2019-05-25 00:56:00.000       SUVI    flux   GOES     NOAA              16     2      284.0
2019-05-25 00:52:00.000 2019-05-25 00:56:00.000       SUVI    flux   GOES     NOAA              16     2      304.0


these are the wavelegths here.

Reply:
Sure, we should check that in the test.

Reply:
So as per my understanding to optimize the tests , I have taken two wavelengths in the 

```
@pytest.mark.parametrize(('start', 'end', 'wave', 'level', 'expected_num_files'), [
    ('2019/05/25 00:50', '2019/05/25 00:54', 94, '1b', 6),
    ('2019/05/25 00:50', '2019/05/25 00:54', 304, '2', 1),
])
```

while the wavelengths here are six which creates conflicts 

am I right?